### PR TITLE
Add linters to CI 

### PR DESCRIPTION
**Pull request by tilne**
* Enable dependency updates via dependabot PRs
* Add semgrep, bandit, and dependency-check to CI
* Add a [pre-commit](https://pre-commit.com/) config file that can be used to automatically run linters pre-commit on modified source

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rexcsn — DISMISSED** [review]

File: .isort.cfg [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [settings]
 line_length=120
 known_future_library=future
```
Comment (rexcsn) [inline .isort.cfg:3]: Probably don't need this anymore?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: ParallelCluster CI
```
Comment (demartinofra) [inline .github/workflows/ci.yml:1]: nice! shall we remove travis config?

Comment (tilne) [reply .github/workflows/ci.yml:1]: From the repo settings in the github UI? I'm planning on doing so after merging these PRs.

File: .isort.cfg (hunk 1 above)
Comment (tilne) [reply .isort.cfg]: removed

**Review by rexcsn — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (demartinofra) [reply .github/workflows/ci.yml:1]: I was referring to the .travis.yml file

Comment (tilne) [reply .github/workflows/ci.yml:1]: Done in #287

Comment (tilne) [reply .github/workflows/ci.yml:1]: Also did this for the cookbook repo in [#910](https://github.com/aws/aws-parallelcluster-cookbook/pull/910)

**Pull request merged**